src/api/dirmonitor/polling.c

```diff
@@ -0,0 +1,123 @@
+#include <stdlib.h>
+#include <string.h>
+#include <fcntl.h>
+#include <unistd.h>
+#include <sys/stat.h>
+#include <SDL.h>
+
+enum {
+  EVENT_MODIFIED
+};
+
+struct file_info {
+  int fd;
+  struct stat file_stat;
```

Comment:
How big is the struct?
Would it make sense to only store `st_mtime`?

Reply:
its fairly chunky, probably best to convert to a time_t